### ISSUE TITLE
OpenVMS host support

### ISSUE THREAD

**Issue by lenticularis39** [lenticularis39/axpbox#78]
OpenVMS host support was removed in the beginning of AXPbox development, mostly because I didn't have any OpenVMS-capable hardware. However that has changed now, and porting the code to OpenVMS shouldn't be that hard, at least in theory that is.

**Comment by RaymiiOrg** [comment]
You have an itanium box? Cool! Would you share some details on the hardware?

Does cmake and a modern (c++ 17) compiler work on OpenVMS?

**Comment by TheFausap** [comment]
@RaymiiOrg  you can wait for the x86 version and they should work fine :) 
I suppose most of the porting issues should disappear as soon as the x86 version will be out for the community. Itanium will be, of course, supported but with x86 support, OpenVMS could become a quite interesting alternative to Linux and for this reason

**Comment by mvorl** [comment]
> You have an itanium box? Cool! Would you share some details on the hardware?
> 
> Does cmake and a modern (c++ 17) compiler work on OpenVMS?

I _think_ I've seen talks of a cmake porting project somewhere.

But the current version of the HP C++ compiler doesn't even fully implement C++03, let alone anything more modern. 20 years of neglect of the platform show everywhere...

**Comment by lenticularis39** [comment]
> You have an itanium box? Cool! Would you share some details on the hardware?

I have an HP Integrity rx2620 with a single 1.3 GHz Intel Itanium 2 CPU, 8 GB of RAM and three SCSI hard drives. It's quite fast and runs OpenVMS fine.

> Does cmake and a modern (c++ 17) compiler work on OpenVMS?

The documentation to the HP C++ compiler is not clear, so I don't know - if it at least has C++11 features, the C++17 specific functions could be implemented as part of the source code. There are signs of OpenVMS support in the CMake codebase - based on what I found in the mailing lists the OpenVMS developers need it to build LLVM (the AMD64 compilers use LLVM as the backend).

**Comment by lenticularis39** [comment]
> But the current version of the HP C++ compiler doesn't even fully implement C++03, let alone anything more modern. 20 years of neglect of the platform show everywhere...

I hope the AMD64 porting effort will accelerate progress of C++ support, which they'll have to do if they want newer LLVM.

**Comment by TheFausap** [comment]
> > But the current version of the HP C++ compiler doesn't even fully implement C++03, let alone anything more modern. 20 years of neglect of the platform show everywhere...
> 
> I hope the AMD64 porting effort will accelerate progress of C++ support, which they'll have to do if they want newer LLVM.

As far as I know, they are using LLVM to do some porting. Hopefully, in the next weeks, I will have access to a 9.1 field test, and I will try something :)

**Comment by mvorl** [comment]
Yes, VSI bases the x86 VMS compilers on LLVM (and a GEM-to-LLVM converter, GEM being the "old" intermediate format). But currently they have to use an old LLVM version until the C++ part is up to compiling LLVM itself. John Reagan talked about it in a [webinar](https://www.youtube.com/watch?v=AD25V21zsak).

**Comment by TheFausap** [comment]
> Yes, VSI bases the x86 VMS compilers on LLVM (and a GEM-to-LLVM converter, GEM being the "old" intermediate format). But currently they have to use an old LLVM version until the C++ part is up to compiling LLVM itself. John Reagan talked about it in a [webinar](https://www.youtube.com/watch?v=AD25V21zsak).

Yes, exactly, but the path is open now :)
